storage/innobase/include/log0log.h

```diff
@@ -506,7 +506,8 @@ struct log_t{
     /** reads buffer from log file
     @param[in]	offset		offset in log file
     @param[in]	buf		buffer where to read */
-    void read(os_offset_t offset, span<byte> buf);
+    dberr_t MY_ATTRIBUTE((warn_unused_result)) read(os_offset_t offset,
+                                                    span<byte> buf);
```

Comment:
I’d add the attribute in a separate line and define this function inline, because the implementation was simplified:
```c++
    MY_ATTRIBUTE((warn_unused_result))
    /** reads buffer from log file
    @param[in]	offset		offset in log file
    @param[in]	buf		buffer where to read
    @return error code */
    dberr_t read(os_offset_t offset, span<byte> buf)
    { return fd.read(offset, buf); }
```